plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/FilesTable.java

```diff
@@ -107,59 +111,65 @@ private static List<Page> buildPages(ConnectorTableMetadata tableMetadata, Table
         Map<Integer, Type> idToTypeMapping = getIcebergIdToTypeMapping(icebergTable.schema());
+        Map<Integer, String> idToQualifiedNameMapping = TypeUtil.indexNameById(icebergTable.schema().asStruct());
 
         TableScan tableScan = icebergTable.newScan()
                 .useSnapshot(snapshotId)
                 .includeColumnStats();
 
-        tableScan.planFiles().forEach(fileScanTask -> {
-            DataFile dataFile = fileScanTask.file();
-
-            pagesBuilder.beginRow();
-            pagesBuilder.appendInteger(dataFile.content().id());
-            pagesBuilder.appendVarchar(dataFile.path().toString());
-            pagesBuilder.appendVarchar(dataFile.format().name());
-            pagesBuilder.appendBigint(dataFile.recordCount());
-            pagesBuilder.appendBigint(dataFile.fileSizeInBytes());
-            if (checkNonNull(dataFile.columnSizes(), pagesBuilder)) {
-                pagesBuilder.appendIntegerBigintMap(dataFile.columnSizes());
+        try (CloseableIterable<FileScanTask> fileScanTasks = tableScan.planFiles()) {
+            for (FileScanTask fileScanTask : fileScanTasks) {
+                DataFile dataFile = fileScanTask.file();
+
+                pagesBuilder.beginRow();
+                pagesBuilder.appendInteger(dataFile.content().id());
+                pagesBuilder.appendVarchar(dataFile.path().toString());
+                pagesBuilder.appendVarchar(dataFile.format().name());
+                pagesBuilder.appendBigint(dataFile.recordCount());
+                pagesBuilder.appendBigint(dataFile.fileSizeInBytes());
+                if (checkNonNull(dataFile.columnSizes(), pagesBuilder)) {
+                    pagesBuilder.appendVarcharBigintMap(remapToQualifiedFieldName(idToQualifiedNameMapping, dataFile.columnSizes()));
+                }
+                if (checkNonNull(dataFile.valueCounts(), pagesBuilder)) {
+                    pagesBuilder.appendVarcharBigintMap(remapToQualifiedFieldName(idToQualifiedNameMapping, dataFile.valueCounts()));
+                }
+                if (checkNonNull(dataFile.nullValueCounts(), pagesBuilder)) {
+                    pagesBuilder.appendVarcharBigintMap(remapToQualifiedFieldName(idToQualifiedNameMapping, dataFile.nullValueCounts()));
+                }
+                if (checkNonNull(dataFile.nanValueCounts(), pagesBuilder)) {
+                    pagesBuilder.appendVarcharBigintMap(remapToQualifiedFieldName(idToQualifiedNameMapping, dataFile.nanValueCounts()));
+                }
+                if (checkNonNull(dataFile.lowerBounds(), pagesBuilder)) {
+                    pagesBuilder.appendVarcharVarcharMap(dataFile.lowerBounds().entrySet().stream()
+                            .filter(entry -> idToTypeMapping.containsKey(entry.getKey()))
+                            .collect(toImmutableMap(
+                                    entry -> idToQualifiedNameMapping.get(entry.getKey()),
```

Comment:
We can create uppercase column names in Spark, but Trino lowercase them when showing. It would be better to lowercase for returning the same column names as Trino table definition in my opinion. Please note that filed names in `row` type already supports uppercase in Trino.

Reply:
Coercing is not mandatory. Please add the product test.